pkg/api/types.go

```diff
@@ -1849,8 +1849,23 @@ func (f *FeatureFlags) IsFeatureEnabled(feature string) bool {
 //for example: if the target is the public azure, then the default container image url should be k8s.gcr.io/...
 //if the target is azure china, then the default container image should be mirror.azure.cn:5000/google_container/...
 func (cs *ContainerService) GetCloudSpecConfig() AzureEnvironmentSpecConfig {
-	targetEnv := helpers.GetTargetEnv(cs.Location, cs.Properties.GetCustomCloudName())
-	return AzureCloudSpecEnvMap[targetEnv]
+	var customCloudName string
+	if cs.Properties != nil {
+		customCloudName = cs.Properties.GetCustomCloudName()
+	}
+	targetEnv := helpers.GetTargetEnv(cs.Location, customCloudName)
+	spec := AzureCloudSpecEnvMap[targetEnv]
+	// TODO Implement ACR-specific overrides below for public cloud
+	/*
+		if spec.CloudName == AzurePublicCloud {
```

Comment:
let's remove this sample code since we most likely don't want to do it this way as discussed in the meeting about this PR (and instead use a function or map).

Reply:
(or change the sample commented code to something like: 

`spec.KubernetesSpecConfig.HyperkubeImageBase = cs.Properties.GetHyperkubeImageBase()`

and then implement the image logic in its own func.

)